Week3/homework/step2-6.js

```diff
@@ -4,13 +4,17 @@ const arr2d = [[1, 2], [3, 4], [5, 6]];
 const arr3d = [[[1, 2], [3, 4]], [[5, 6], [7, 8]]];
 
 function flattenArray2d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  return arr
+    .toString()
+    .split(',')
+    .map(element => +element);
 }
 
 function flattenArray3d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  return arr
+    .toString()
+    .split(',')
+    .map(element => +element);
```

Comment:
Interesting! I wouldn't have ever thought to use `toString` to flatten the array. I guess this would work with any level of nested arrays too, so it's a very creative solution.

This does have limitations though - for instance it wouldn't work if one of the items in the array happened to be a function, and it might have performance penalties for larger arrays.

@yash-kapila curious on your thoughts here too?

Reply:
Thank you for your time and nice comment. I will try to fix it with other solutions.